theano/tests/unittest_tools.py

```diff
@@ -244,9 +244,9 @@ def _compile_and_check(self, inputs, outputs, numeric_inputs, cls,
         # Check that the Op is removed from the compiled function.
         if check_topo:
             topo_shape = shapes_function.maker.fgraph.toposort()
+            topo_out = outputs_function.maker.fgraph.toposort()
             assert not any(isinstance(t.op, cls) for t in topo_shape)
-        topo_out = outputs_function.maker.fgraph.toposort()
-        assert any(isinstance(t.op, cls) for t in topo_out)
```

Comment:
I don't absorb this code very well. In the docstring of that function, it says `check_topo` flags whether to check the Op was removed from the graph. So assumed those 2 lines *should* be under the if clause. Without doing so it would cause test error since OfG node was removed due to inline. Also this doesn't change default behavior since `check_topo=True` for default. Am I doing anything wrong?